src/include/mallocMC/creationPolicies/Scatter_impl.hpp

```diff
@@ -939,8 +1003,24 @@ namespace mallocMC
                         "Maybe you are using an incompatible ReservePoolPolicy "
                         "or AlignmentPolicy.");
                 }
-                ScatterKernelDetail::initKernel<<<1, 256>>>(
-                    heap, pool, memsize);
+                auto initKernel = [] ALPAKA_FN_ACC(
+                                      const AlpakaAcc & acc,
+                                      T_DeviceAllocator * heap,
+                                      void * heapmem,
+                                      size_t memsize) {
+                    heap->pool = heapmem;
+                    heap->initDeviceFunction(acc, heapmem, memsize);
+                };
+                // TODO(bgruber): how do I know which dimension the user's
+                // AlpakaAcc is using?
+                using Dim = alpaka::dim::traits::DimType<AlpakaAcc>::type;
+                using Idx = alpaka::idx::traits::IdxType<AlpakaAcc>::type;
+                const auto workDiv = alpaka::workdiv::WorkDivMembers<Dim, Idx>{
+                    Idx{1}, Idx{256}, Idx{1}};
```

Comment:
I do not think it works. I believe the way is to provide three values of type `alpaka::vec::Vec<Dim, Idx>` to the `workDiv ` constructor. And such a vector can't be constructed from one value in 2D and 3D. A possible, but clunky way to do that is to get a vector of 1 using its static method `ones()` and then set one of its elements to what you want.

Reply:
It works btw! If I specify a workDiv with a lower dimension, the other dimensions are set to 1, which seems reasonable to me. What is less intuitive now is how you get the index and worksize in the kernel:
```
const auto id = alpaka::idx::getIdx<alpaka::Grid, alpaka::Threads>(acc).sum();
const auto n = alpaka::workdiv::getWorkDiv<alpaka::Grid, alpaka::Threads>(acc).prod();
```
This should work, because the index contains 0 for all unused dimensions, and the workDiv contains 1 for all unused dimensions.
I would have liked to just use `[0]` from index and workDiv, but unfortunately alpaka indices are stored `[x]`, `[y, x]` or `[z, y, x]` and checking the dimension and indexing is way more code than `sum()` and `prod()`.